iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/DanglingLineXml.java

```diff
@@ -70,26 +97,106 @@ protected DanglingLine readRootElementAttributes(DanglingLineAdder adder, Networ
+        if (!Double.isNaN(permanentLimit[0]) || !temporaryLimits.isEmpty()) {
+            CurrentLimitsAdder limitsAdder = danglingLine[0].newCurrentLimits()
+                    .setPermanentLimit(permanentLimit[0]);
+            temporaryLimits
+                    .forEach((acceptableDuration, tl) -> limitsAdder.beginTemporaryLimit()
+                    .setAcceptableDuration(acceptableDuration)
+                    .setName(tl.name)
+                    .setValue(tl.value)
+                    .setFictitious(tl.fictitious)
+                    .endTemporaryLimit());
+            limitsAdder.add();
+        }
     }
 
-    @Override
-    protected void readSubElements(DanglingLine dl, NetworkXmlReaderContext context) throws XMLStreamException {
-        readUntilEndRootElement(context.getReader(), () -> {
-            if ("currentLimits".equals(context.getReader().getLocalName())) {
-                readCurrentLimits(null, dl::newCurrentLimits, context.getReader());
-            } else {
-                super.readSubElements(dl, context);
+    private DanglingLine readGeneration(DanglingLineAdder adder, XMLStreamReader reader, NetworkXmlReaderContext context) throws XMLStreamException {
+        double minP = XmlUtil.readOptionalDoubleAttribute(reader, "minP");
+        double maxP = XmlUtil.readOptionalDoubleAttribute(reader, "maxP");
+        boolean voltageRegulationOn = XmlUtil.readBoolAttribute(reader, "voltageRegulationOn");
+        double targetP = XmlUtil.readOptionalDoubleAttribute(reader, "targetP");
+        double targetV = XmlUtil.readOptionalDoubleAttribute(reader, "targetV");
+        double targetQ = XmlUtil.readOptionalDoubleAttribute(reader, "targetQ");
+        adder.newGeneration()
+                .setMinP(minP)
+                .setMaxP(maxP)
+                .setVoltageRegulationOn(voltageRegulationOn)
+                .setTargetP(targetP)
+                .setTargetV(targetV)
+                .setTargetQ(targetQ)
+                .add();
+        DanglingLine danglingLine = adder.add();
+        XmlUtil.readUntilEndElement(GENERATION, reader, () -> {
+            switch (context.getReader().getLocalName()) {
+                case "reactiveCapabilityCurve":
+                case "minMaxReactiveLimits":
+                    ReactiveLimitsXml.INSTANCE.read(danglingLine.getGeneration(), context);
+                    break;
+                default:
+                    throw new PowsyblException("Unknown element name <" + context.getReader().getLocalName() + "> in <" + danglingLine.getId() + ".generation>");
             }
         });
+        return danglingLine;
+    }
+
+    class TemporaryLimitXml {
+        private final String name;
+        private final double value;
+        private final boolean fictitious;
+
+        TemporaryLimitXml(String name, double value, boolean fictitious) {
+            this.name = name;
+            this.value = value;
+            this.fictitious = fictitious;
+        }
```

Comment:
I'm surprised: this piece of code change a lot. To be checked, but maybe there is a design issue that makes this code complex.

Reply:
It changes a lot because by moving `GenerationAdder` in `DanglingLineAdder`, we now cannot call `add()` before reading sub elements (similar to what is done in `ShuntXml`)

Reply:
Do you prefer using attributes in danglingline like for threewindingstransformers?